weatherman.py

```diff
@@ -0,0 +1,189 @@
+import calendar
+import csv
+import os
+import sys
+
+
+class temperatureTracker():
+
+    def __init__(self):       
+        self.max_temperature = float('-inf')
+        self.max_temperature_date = ""
+        self.min_temperature = float('inf')
+        self.min_temperature_date = ""
+        self.max_humidity = float('-inf')
+        self.max_humidity_date = ""
+    
+    def set_max_temperature(self, temperature, date):
+        self.max_temperature = temperature
+        self.max_temperature_date = date
+
+    def set_min_temperature(self, temperature, date):
+        self.min_temperature = temperature
+        self.min_temperature_date = date
+    
+    def set_max_humidity(self, humidity, date):
+        self.max_humidity = humidity
+        self.max_humidity_date = date
+    
+    def get_max_temps(self):
+        return self.max_temperature, self.min_temperature, self.max_humidity
+
+    def print_temperatures_year(self):
+        print(f"highest temperature is: {self.max_temperature} C on {self.max_temperature_date}")
+        print(f"lowest temperature is: {self.min_temperature} C on {self.min_temperature_date}")
+        print(f"highest humidity percentage is: {self.max_humidity} % on {self.max_humidity_date}")
+
+
+temperature_tracker = temperatureTracker()
+
+
+def open_csv_file(file_name):    
+    csv_file = open(file_name) 
+    return csv_file
+
+
+def read_weather_data_from_csv(csv_file):
+    weather_data = list(csv.DictReader(csv_file, skipinitialspace=True))
+    if len(weather_data) == 0:
+        return False
+    return weather_data
+
+
+def match_file_name(year, month_name):                            
+    flag = False    
+    for file_name in os.listdir('.'):
+        if year in file_name and month_name in file_name:
+            return file_name          
+    if not flag:     
+        print("NO RECORD FOUND AGAINT THIS NAME")
+        return False
+
+
+def extract_int_value(value): return int(value) if value != "" else 0
```

Comment:
Functions are not supposed to converted into one liners. Only return statement can be one liner down the definition.